R/calcFEbyEUEC.R

```diff
@@ -134,66 +134,33 @@ calcFEbyEUEC <- function() {
   # existing disaggregated data replaces values from optimization
   data <- ieaIODis %>%
     left_join(dataReplaceFull, by = c("region", "period", "carrier", "enduse")) %>%
-    group_by(across(all_of(c("region", "period", "carrier")))) %>%
     mutate(valueUncorrected = ifelse(is.na(.data[["replaceValue"]]),
                                      .data[["value"]],
-                                     .data[["replaceValue"]]),
-           value = .data[["valueUncorrected"]] * .data[["value"]] / sum(.data[["value"]])) %>%
+                                     .data[["replaceValue"]])) %>%
+    group_by(across(all_of(c("region", "period", "carrier")))) %>%
+    mutate(value = ifelse(.data[["valueUncorrected"]] == 0,
+                          0,
+                          sum(.data[["value"]]) * proportions(.data[["valueUncorrected"]]))) %>%
+    ungroup() %>%
     select(-"valueUncorrected") %>%
     select("region", "period", "unit", "carrier", "enduse", "value")
 
 
   # CORRECTIONS ----------------------------------------------------------------
 
-  # For unknown reasons, the enduse share of "space_cooling" for region "Africa"
-  # is not met and will therefore be corrected. Since "space_cooling" only corresponds
-  # to the carrier "elec", the correction is straight-forward.
-  # TODO: check if this can be fixed #nolint
-
-  elecSpaceCoolingShare <- sharesEU %>%
-    filter(.data[["region"]] == "Africa",
-           .data[["enduse"]] == "space_cooling") %>%
-    select("period", "value") %>%
-    rename("share" = "value")
-
-  dataCorr <- data %>%
-    left_join(regmapping, by = c("region")) %>%
-    filter(.data[["regionAgg"]] == "Africa") %>%
-    left_join(elecSpaceCoolingShare, by = c("period")) %>%
-    select(-"regionAgg") %>%
-    group_by(across(-all_of(c("enduse", "carrier", "share", "value")))) %>%
-    mutate(value = ifelse(.data[["enduse"]] == "space_cooling",
-                          ifelse(.data[["carrier"]] == "elec",
-                                 sum(.data[["value"]], na.rm = TRUE) * .data[["share"]],
-                                 0),
-                          .data[["value"]] * (1 - .data[["share"]])),
-           value = .data[["value"]] * ifelse(.data[["enduse"]] == "space_cooling",
-                                             1,
-                                             .data[["share"]] * sum(.data[["value"]]) /
-                                               sum(.data[["value"]][.data[["enduse"]] == "space_cooling"]))) %>%
-    ungroup() %>%
-    select(-"share")
-
-  dataFull <- rbind(dataCorr,
-                    data %>%
-                      left_join(regmapping, by = c("region")) %>%
-                      filter(.data[["regionAgg"]] != "Africa") %>%
-                      select(-"regionAgg"))
-
-
   # Since the data on district cooling is very sparse and the low global penetration
   # of the technology, we assume that all historic cooling demand is covered by
```

Comment:
maybe also change this to "... and the technology exhibits a low global penetration rate, ..."

Reply:
Thanks, I adopted that.